src/main/java/org/spongepowered/api/entity/ai/task/AITask.java

```diff
@@ -75,7 +75,7 @@
      * @param other The other task
      * @return True if it can be, false if not
      */
-    boolean canRunConcurrentWith(AITask<O> other);
+    boolean canRunConcurrentWith(AITask<? super O> other);
```

Comment:
Because in the current impl you use that many casts that the usefullness is gone anyways. 
The issue is not the single entry but the final stored list. I dont have an example ready but its quite obvious when using it.